packages/fetchai/skills/weather_client/handlers.py

```diff
@@ -125,6 +122,7 @@ def _handle_propose(self, msg: FipaMessage, dialogue: Dialogue) -> None:
         :param dialogue: the dialogue object
         :return: None
         """
+        dialogues = cast(Dialogues, self.context.dialogues)
```

Comment:
remove

Reply:
Need a small change for this to work. The `dialogue.update` now misses the `update_self_initiated_dialogue_label_on_second_message` functionality in `dialogues.update()` 

I will add this in the next PR where stuff are moved into base.